src/sentry/utils/sdk.py

```diff
@@ -223,7 +223,16 @@ def before_send_transaction(event: Event, _: Hint) -> Event | None:
         return None
 
     # Occasionally the span limit is hit and we drop spans from transactions, this helps find transactions where this occurs.
-    num_of_spans = len(event["spans"])
+    if isinstance(event["spans"], AnnotatedValue):
+        # AnnotatedValue isn't generic so we check its inner value's type otherwise mypy will
+        # complain. The TypeError should be unreachable.
+        if isinstance(event["spans"].value, Sized):
```

Comment:
We could probably make `AnnotatedValue` generic in the SDK, if that would help you here

Reply:
Yes please.

Reply:
And perhaps expose it publicly as well, so you don't have to access it through the `_types` module (which is meant to be internal to the SDK, and subject to change in minor releases)

Reply:
@cmanallen [issue here](https://github.com/getsentry/sentry-python/issues/4077)

Reply:
Thanks!